### PR TITLE
fixed a bunch of errors found when testing the code, refactored endpoint responses, tested all endpoints

### PR DESCRIPTION
**Pull request by AlexLam1997**

### CODE REVIEW

File: backend/logic/courses/courses.js [hunk 1]
```diff
@@ -373,9 +344,8 @@ var assignCourseToCurriculum = async (courseType, courseCode, curriculum) => {
       [curriculum]
     );
   } catch (err) {
-    console.log(err);
     conn.release();
-    throw Error("Internal Server Error!\n");
```
Comment (KevenLYF) [inline backend/logic/courses/courses.js]: I'm using this string to classify 500 (Internal error) , how do you know if it's an internal error if remove it? I checked the `customeResponse` didn't see any 500 handler, did I miss something or ?

File: backend/logic/courses/courses.js [hunk 2]
```diff
@@ -38,37 +34,30 @@ var queryCourseByTag = async function queryCourseByTag(tag) {
...
-    ]);
-    return true;
-  } catch (error) {
-    connection.rollback();
-    console.error(error);
-    throw new Error(false);
-  } finally {
-    connection.release();
-  }
+  await connection.query("INSERT INTO courses VALUES(?, ?, ?, ?);", [
```
Comment (si-x-li) [inline backend/logic/courses/courses.js]: Add column fields to the query

File: backend/logic/users/users.js [hunk 3]
```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
 
   try {
     await connection.beginTransaction();
-    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
-    await connection.query("INSERT INTO students VALUES(?, ?);", [id, username]);
+    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [
```
Comment (si-x-li) [inline backend/logic/users/users.js]: Add column values to remain consistent with other queries

File: backend/logic/users/users.js [hunk 4]
```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
...
+      username,
+      email,
+      hash
+    ]);
+    await connection.query("INSERT INTO students VALUES(?, ?);", [
```
Comment (si-x-li) [inline backend/logic/users/users.js]: Same as above

File: backend/logic/users/users.js [hunk 5]
```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
...
 
   let connection = await mysql.getNewConnection();
 
   // Hash the password
   let hash = hasher.hashPass(password);
   try {
     await connection.beginTransaction();
-    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
-    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [id, username]);
+    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [
```
Comment (si-x-li) [inline backend/logic/users/users.js]: Same here

File: backend/logic/users/users.js [hunk 6]
```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
...
+      username,
+      email,
+      hash
+    ]);
+    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [
```
Comment (si-x-li) [inline backend/logic/users/users.js]: Same here

File: backend/logic/users/users.js [hunk 7]
```diff
@@ -29,157 +17,128 @@ var insertStudentUser = async (username, password, email, id) => {
...
 
   try {
     await connection.beginTransaction();
-    let staff_id = await connection.query("SELECT staff_id FROM staff_members WHERE username = ?;", username);
-    await connection.query("DELETE FROM staff_members WHERE staff_id = ?;", [staff_id[0].staff_id]);
+    let staff_id = await connection.query(
+      "SELECT staff_id FROM staff_members WHERE username = ?;",
+      username
+    );
+    await connection.query("DELETE FROM staff_members WHERE staff_id = ?;", [
```
Comment (si-x-li) [inline backend/logic/users/users.js]: Pretty sure we're able to reduce these 2 queries into a single one without causing any issues

DELETE FROM staff_members WHERE staff_id IN (SELECT staff_id FROM staff_members WHERE username = ?);

File: backend/logic/courses/courses.js (hunk 1 above)
Comment (AlexLam1997) [reply backend/logic/courses/courses.js]: It defaults to sending back a 500 error code if no error code is specified. If you want to specify a specific code you can do so with: 
```
let error = new Error ("message") 
error.code = 400; 
throw error; 
```

**Pull request merged**